ckanext/scheming/templates/scheming/package/snippets/resource_form.html

```diff
@@ -47,34 +47,7 @@
   {%- endif -%}
 
   {%- set schema = h.scheming_get_dataset_schema(dataset_type) -%}
-  {%- for field in schema.resource_fields -%}
-    {%- if field.form_snippet is not none -%}
-      {%- if field.field_name not in data %}
-        {# Set the field default value before rendering but only if
-        it doesn't already exist in data which would mean the form
-        has been submitted. #}
-        {% if field.default_jinja2 %}
-          {% do data.__setitem__(
-            field.field_name,
-            h.scheming_render_from_string(field.default_jinja2)) %}
-        {% elif field.default %}
-          {% do data.__setitem__(field.field_name, field.default) %}
-        {% endif %}
-      {% endif -%}
-      {# We pass pkg_name as the package_id because that's the only
-      variable available in this snippet #}
-      {%- snippet 'scheming/snippets/form_field.html',
-        field=field,
-        data=data,
-        errors=errors,
-        licenses=c.licenses,
-        entity_type='dataset',
-        object_type=dataset_type,
-        package_id=pkg_name
-      -%}
-    {%- endif -%}
-  {%- endfor -%}
-
+  {% snippet 'scheming/snippets/render_fields.html', fields=schema.resource_fields, data=data, errors=errors, entity_type='dataset', object_type=dataset_type, set_fields_defaults=true %}
```

Comment:
entity_type='resource'?

Reply:
I don't know, just copied it from the old template. If you ask me, the code must be refactored to remove  ` licenses=licenses, entity_type=entity_type, object_type=object_type` from here.